src/actions/cart-manager.ts

```diff
@@ -0,0 +1,92 @@
+import {retrieveSpecificProduct} from './DAL/product-db-actions';
+
+export const mainCart: { id: string, users: { userId: string, amount: number }[] }[] = [];
+
+export const addToCart = async (ip: string, productId: string, desiredAmount: number) => {
+    const product = await retrieveSpecificProduct(productId);
+    let availableAmount;
+    if (!product)
```

Comment:
use `{` and `{` in `if`//`else`/etc....
BTW add tslint to the project - will fix you all those errors which I hate

Reply:
I have tslint

Reply:
add `"tslint-eslint-rules",` to extends - will help you with those

Reply:
done